strawberryfields/engine.py

```diff
@@ -531,161 +544,541 @@ def run(self, program, *, args=None, compile_options=None, run_options=None):
+            self.__class__.__name__, self.token, self.host
+        )
 
-        # Job either failed or is complete - in either case, clear the job queue so that the engine is
-        # ready for future jobs.
-        self.reset()
+    def __str__(self):
+        return self.__repr__()
 
-        if job.is_failed:
-            message = str(job.manager.http_response_data["meta"])
-            raise JobExecutionError(message)
-        elif job.is_complete:
-            job.result.manager.get()
-            return job.result.result.value
 
-    def run(self, program, shots=1, **kwargs):
-        """Compile the given program and execute it by queuing a job in the Starship.
+class StarshipEngine:
+    """A quantum program executor engine that that provides a simple interface for
+    running remote jobs in a synchronous or asynchronous manner.
 
-        For the :class:`Program` instance given as input, the following happens:
+    **Example:**
 
-        * The Program instance is compiled for the target backend.
-        * The compiled program is sent as a job to the Starship
-        * The measurement results of each subsystem (if any) are stored in the :attr:`~.BaseEngine.samples`.
-        * The compiled program is appended to self.run_progs.
-        * The queued or completed jobs are appended to self.jobs.
+    The following example instantiates an engine with the default configuration, and
+    runs jobs both synchronously and asynchronously.
 
-        Finally, the result of the computation is returned.
+    .. code-block:: python
+
+        engine = StarshipEngine("chip2")
+
+        # Run a job synchronously
+        result = engine.run(program, shots=1)
+        # (Engine blocks until job is complete)
+        result  # [[0 1 0 2 1 0 0 0]]
+
+        # Run a job synchronously, but cancel it before it is completed
+        result = engine.run(program, shots=1)
+        ^C # KeyboardInterrupt cancels the job
+
+        # Run a job asynchronously
+        job = engine.run_async(program, shots=1)
+        job.status  # <JobStatus.QUEUED: 'queued'>
+        job.result  # InvalidJobOperationError
+        # (After some time...)
+        job.refresh()
+        job.status  # <JobStatus.COMPLETE: 'complete'>
+        job.result  # [[0 1 0 2 1 0 0 0]]
+
+    Args:
+        target (str): the target device
+        connection (strawberryfields.engine.Connection): a connection to the remote job
+            execution platform
+    """
+
+    POLLING_INTERVAL_SECONDS = 1
+    VALID_TARGETS = ("chip2",)
+
+    def __init__(self, target: str, connection: Connection = None):
+        if target not in self.VALID_TARGETS:
+            raise ValueError(
+                "Invalid engine target: {} (valid targets: {})".format(
+                    target, self.VALID_TARGETS
+                )
+            )
+        if connection is None:
+            # TODO use `load_config` when implemented
+            config = DEFAULT_CONFIG["api"]
```

Comment:
This line should be updated to use `load_config()` after https://github.com/XanaduAI/strawberryfields/pull/298 is merged.

Reply:
It got merged just now :)

Reply:
Great, I've updated the relevant logic to use the new configuration functions!